### ISSUE TITLE
Android App

### ISSUE THREAD

**Issue by Jeffrey131313** [MCCTeam/Minecraft-Console-Client#2145]
### Prerequisites

- [X] I have read and understood the [user manual](https://github.com/MCCTeam/Minecraft-Console-Client/tree/master/MinecraftClient/config)
- [X] I made sure I am running the latest [development build](https://ci.appveyor.com/project/ORelio/Minecraft-Console-Client/build/artifacts)
- [X] I tried to [look for similar feature requests](https://github.com/MCCTeam/Minecraft-Console-Client/issues?q=is%3Aissue) before opening a new one

### Console Client Version

new

### Describe your problem

Can you make it into an app? Because usually we don't take the computer with us when we are out

### Device

Desktop

### Operating System

Windows

**Comment by breadbyte** [comment]
You can install [Termux](https://play.google.com/store/apps/details?id=com.termux&hl=en&gl=US) on Android and run the application that way.

**Comment by Jeffrey131313** [comment]
> 您可以在 Android 上安装[Termux](https://play.google.com/store/apps/details?id=com.termux&hl=en&gl=US)并以这种方式运行应用程序。
I think it's too troublesome, but thank you for your idea

**Comment by milutinke** [comment]
I just happen to have tried setting up MCC on Android using Termux and Andronix, but the problem is with .NET 6 not working properly on older distros on ARM64, since they're all outdated in Andronix.
I'll try once more, but with pre-built versions.
My plan was to try it out, figure the easiest/best/quickest way and then write a page in the documentation.
Porting this into an app would be really complicated, but there will be a solution soon, I'm working on it, for people who do not want to mess with Termux.
We're going to use a WebSocket Chat Bot, I'm currently developing it: https://github.com/MCCTeam/Minecraft-Console-Client/pull/2126
Since WebSocket can be used in a web browser, I can make an app for Android and iOS using JavaScript.

**Comment by Jeffrey131313** [comment]
thank you

**Comment by Splarkszter** [comment]
Huh. never thought about this. Making this work could leave free more of my potato APU to render the game. XD

**Comment by Splarkszter** [comment]
> I just happen to have tried setting up MCC on Android using Termux and Andronix, but the problem is with .NET 6 compilation not working on older distros on ARM64, since they're all outdated in Andronix. I'll try once more, but with pre-built versions. My plan was to try it out, figure the easiest/best/quickest way and then write a page in the documentation. Porting this into an app would be really complicated, but there will be a solution soon, I'm working on it, for people who do not want to mess with Termux. We're going to use a WebSocket Chat Bot, I'm currently developing it: #2126 Since WebSocket can be used in a web browser, I can make an app for Android and iOS using JavaScript.

I found this on a quick search on google https://dotnet.microsoft.com/en-us/apps/xamarin

**Comment by milutinke** [comment]
> > I just happen to have tried setting up MCC on Android using Termux and Andronix, but the problem is with .NET 6 compilation not working on older distros on ARM64, since they're all outdated in Andronix. I'll try once more, but with pre-built versions. My plan was to try it out, figure the easiest/best/quickest way and then write a page in the documentation. Porting this into an app would be really complicated, but there will be a solution soon, I'm working on it, for people who do not want to mess with Termux. We're going to use a WebSocket Chat Bot, I'm currently developing it: #2126 Since WebSocket can be used in a web browser, I can make an app for Android and iOS using JavaScript.
> 
> I found this on a quick search on google https://dotnet.microsoft.com/en-us/apps/xamarin

I know about it already, it will require a lot of code.
I think none of us here from the maintainer/contributor team is familiar with Xamarin.
If someone is, feel free to give it a try.

**Comment by milutinke** [comment]
On our part, we could do the abstraction part, that would enable for MCC to be embedded, which would open a lot of possibilities.

**Comment by milutinke** [comment]
Yoooo, finally got it working on Termux.
I'll update the documentation tomorrow.

![Screenshot_20220906-011442_Termux.jpg](https://user-images.githubusercontent.com/441903/188519732-3f7e7698-9449-4908-b8a6-d54456e5aadc.jpg)

**Comment by Splarkszter** [comment]
nice!!!!

**Comment by milutinke** [comment]
The documentation section is finished
https://mccteam.github.io/docs/guide/installation.html#running-on-android

**Comment by milutinke** [comment]
> > For iPhone/iOS users, there is [iSH](https://github.com/ish-app/ish), sadly I do not own an iPhone so I can't test it out.
> > If someone who has an iPhone and some time, let's figure out the process for it and make a documentation page.
> > Installation of iSH with Alpine linux: https://www.youtube.com/watch?v=wIGVLM3Oaag
> 
> iSH is running on 32 bit system. I don't think it is possible to run MCC, or even install .NET on it. ![image](https://user-images.githubusercontent.com/39955851/188672539-ae46489a-31bc-4a26-8887-965ac6af0e84.jpeg)

There is a 32 bit ARM build for Alpine: https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-aspnetcore-6.0.8-linux-arm32-alpine-binaries
If I am not mistaken, iPhones have ARM processors?

**Comment by ReinforceZwei** [comment]
Yes, iPhone is using ARM CPU, but iSH is not. iSH is using x86 architecture.

**Comment by Splarkszter** [comment]
> The documentation section is finished https://mccteam.github.io/docs/guide/installation.html#running-on-android


ahhh yes. that's how a good doc is done. Excelent work!!!, i'm gonna follow it and share the results.

**Comment by Splarkszter** [comment]
Ok, so i had problems with F-Droid install button was not there, pretty unreliable then, because some issue posts said F-Droid is very slow for updates and fixes, so it's just best to go to the original repository of the App. (Which is this one https://github.com/termux/termux-app#github ). And then with an image point or highlight the releases button inside on GitHub.

On the step when you run ubuntu there is a typo in the command, this one saying './startubunUT.sh' instead of './startubunTU.sh'

Other than that i seems to be running correctly. i'll test modifying the config file later for remote control and scripts.

**Comment by milutinke** [comment]
> Ok, so i had problems with F-Droid install button was not there, pretty unreliable then, because some issue posts said F-Droid is very slow for updates and fixes, so it's just best to go to the original repository of the App. (Which is this one https://github.com/termux/termux-app#github ). And then with an image point or highlight the releases button inside on GitHub.
> 
> On the step when you run ubuntu there is a typo in the command, this one saying './startubunUT.sh' instead of './startubunTU.sh'
> 
> Other than that i seems to be running correctly. i'll test modifying the config file later for remote control and scripts.
> 
> Note: My device is a 'Samsung Galaxy S9'

I'll update the documentation to use the git repo instead of F-Droid.
For me it worked, sometimes the install button needs some time to appear, that might be the case on your phone.
I'm using Samsung Galaxy A9 2018.

**Comment by milutinke** [comment]
> why is ubuntu or dotnet needed for a termux installation? can't i use mcc without those? in offline mode maybe?

@KoyunMaybeHere The MCC cannot work without .net because it's built using it.
I've updated the tutorial in the documentation to simplify the installation process.

I'll close this issue as the Android app won't be made without the project being refactored first.
